lib/type-registry.js

```diff
@@ -0,0 +1,118 @@
+// Copyright IBM Corp. 2016. All Rights Reserved.
+// Node module: strong-remoting
+// This file is licensed under the Artistic License 2.0.
+// License text available at https://opensource.org/licenses/Artistic-2.0
+
+'use strict';
+
+var ArrayConverter = require('./types/array');
+var assert = require('assert');
+var debug = require('debug')('strong-remoting:types');
+var g = require('strong-globalize')();
+
+module.exports = TypeRegistry;
+
+function TypeRegistry(options) {
+  this._options = options || {};
+  this._types = Object.create(null);
+  this._registerBuiltinTypes();
+}
+
+TypeRegistry.prototype.registerType = function(typeName, converter) {
+  assert(typeof typeName === 'string' && typeName,
+    'typeName must be a non-empty string');
+  assert(typeof converter === 'object' && converter,
+    'converter must be an object');
```

Comment:
Are you just checking the type of `converter`. In this case shouldn't be like the following?
```
  assert(typeof converter === 'object',
     'converter must be an object');
```